metrics/tempdata/tempdata.go

```diff
@@ -0,0 +1,147 @@
+package tempdata
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/filecoin-project/lassie/pkg/types"
+)
+
+const timeout = 1 * time.Minute
+
+type TempData struct {
+	timer                           *time.Timer
+	lastStage                       uint32
```

Comment:
Considering the go mod version is `1.19` you can simplify code by using `atomic.Uint32` instead of regular types and atomic swap on pointers. Ditto re remaining numerical fields.

Reply:
honestly, I prefer the old atomics at the end of the day :P